cdc/sink/mq.go

```diff
@@ -310,8 +310,13 @@ func (k *mqSink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
 }
 
 func (k *mqSink) Close(ctx context.Context) error {
-	err := k.mqProducer.Close()
-	return errors.Trace(err)
+	closedCh := k.mqProducer.AsyncClose()
+	select {
+	case <-ctx.Done():
```

Comment:
Please add a comment here to tell this `ctx` is a `canceled context` so that this select statement will never block.